src/ui/shared/requests/uiInterpretTransaction.ts

```diff
@@ -0,0 +1,135 @@
+import { hashQueryKey, useQuery } from '@tanstack/react-query';
+import {
+  interpretSignature,
+  interpretTransaction,
+} from 'src/modules/ethereum/transactions/interpret';
+import type { IncomingTransactionWithChainId } from 'src/modules/ethereum/types/IncomingTransaction';
+import { getNetworksStore } from 'src/modules/networks/networks-store.client';
+import { invariant } from 'src/shared/invariant';
+import { normalizeChainId } from 'src/shared/normalizeChainId';
+import {
+  getPreferences,
+  usePreferences,
+} from 'src/ui/features/preferences/usePreferences';
+import { fetchAndAssignPaymaster } from 'src/modules/ethereum/account-abstraction/fetchAndAssignPaymaster';
+import { Client } from 'defi-sdk';
+import { ZerionAPI } from 'src/modules/zerion-api/zerion-api.client';
+import { useDefiSdkClient } from 'src/modules/defi-sdk/useDefiSdkClient';
+import { useCurrency } from 'src/modules/currency/useCurrency';
+import { walletPort } from '../channels';
+
+interface EligibilityQuery {
+  data?: null | { data: { eligible: boolean } };
+  status: 'error' | 'success' | 'loading';
+  isError: boolean;
+}
+
+/**
+ * Interprets Paymaster Transaction if it is eligible
+ * or a regular transaction otherwise
+ */
+export async function interpretTxBasedOnEligibility({
```

Comment:
Feel free to suggest a better name here :)

Reply:
I think, we can go with a simple `interpretTransaction`. Cause ideally, this should just interpret any transaction, everything else is just a details of implementation